src/bacnet/basic/service/h_apdu.c

```diff
@@ -468,7 +468,9 @@ uint16_t apdu_decode_confirmed_service_request(uint8_t *apdu, /* APDU data */
                 return 0;
             }
         }
-        if (apdu_len == (len + 1)) {
+        if (apdu_len > MAX_APDU){
```

Comment:
Please use 'len' instead of MAX_APDU (it should be the same (less the NPDU) when it gets into this function).

Reply:
Thanks Steve. This time  I have to admit that I don't understand how to use len check if apdu_len is larger than MAX_APDU, 1476 bytes in my case.
Your existing code takes care of 
apdu_len >= len+1 but fails the test.

Reply:
npdu_handler(pdu, pdu_len) -> where pdu_len <= MAX_MPDU, and MAX_MPDU = datalink header + MAX_PDU, and MAX_PDU = MAX_NPDU+MAX_APDU.

Hmm, I see the point. The datalink header and NPDU are not fixed sizes.